ls++

```diff
@@ -362,6 +383,9 @@ sub perm_size_file {
 
 sub perm {
   my ($perm) = @_;
+  if (substr($perm,-1,1) ne "+"){
+    $perm .= " "; # adding an extra space to align with occasional +
```

Comment:
I wasn't very clear in my comment, but I meant going through each line of the GNU ls output like you did in the get_max_size_len subroutine, which means looping through each line twice just to get the max length. I thought it would be some extra steps for very little benefit since these "+" mask bits are rather uncommon.
But I'll work on it and add the space only when necessary then.